node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -551,22 +549,36 @@ func (atp *apiTransactionProcessor) getVirtualNonce(address []byte, accountsAdap
 		TransactionsProcessor: &txProcessor,
 	}
 
-	// TODO brainstorm if the selection session is needed
+	// the SelectionSession is used in this flow for fallbacks (e.g. the account does not exist in the proposed blocks, unexpected errors etc.)
 	selectionSession, err := preprocess.NewSelectionSession(argsSelectionSession)
 	if err != nil {
-		log.Warn("apiTransactionProcessor.getVirtualNonce could not create SelectionSession")
-		return 0, err
+		log.Warn("apiTransactionProcessor.getVirtualNonceWithBlockInfo could not create SelectionSession")
+		return 0, nil, err
 	}
 
-	// TODO brainstorm if the blockchain info is needed
-	blockchainInfo := holders.NewBlockchainInfo(nil, nil, 0)
-	virtualNonce, err := txCache.GetVirtualNonce(address, selectionSession, blockchainInfo)
+	// TODO use the right information for blockchainInfo
+	// these blockchainInfo should contain the hash of the last committed block and the actual nonce
+	// these variables will also be used for the response
+	// NOTE: should not remain like this
+	var latestCommittedBlockHash []byte
+	currentNonce := uint64(1)
+
+	blockchainInfo := holders.NewBlockchainInfo(nil, latestCommittedBlockHash, currentNonce)
```

Comment:
The cache, which offers the virtual nonce, must mention the latest committed block hash - since the two values should be correlated, linked. That is, the cache should return the `latest committed block hash` - in its opinion (view), given its tracked blocks.